### PR TITLE
scaling demo

### PR DESCRIPTION
**Pull request by Agwen12**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
         <dependency>
             <groupId>org.k8loud</groupId>
             <artifactId>themis-executor</artifactId>
-            <version>0.0.3.3</version>
+            <version>0.0.4.0</version>
```
Comment (proman3419) [inline pom.xml:27]:
```suggestion
            <version>0.0.4.1</version>
```

File: scenarios/k8s-scaling/cm.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+apiVersion: v1
```
Comment (proman3419) [inline scenarios/k8s-scaling/cm.yml:1]: Isn't it a duplicate of rules.drl? If so remove this file please

File: scenarios/k8s-scaling/rules.drl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+             cronChecker.checkPatternForSession("* */3 * ? * * *"))
+    then
+        kubernetesService = usableServices.getKubernetesService();
+        HorizontalScalingAction.builder()
+                    .kubernetesService(kubernetesService)
+                    .namespace("sock-shop")
+                    .resourceType("Deployment")
+                    .resourceName("front-end")
+                    .replicas(1)
+                    .build();
```
Comment (proman3419) [inline scenarios/k8s-scaling/rules.drl:25]:
```suggestion
                    .build()
                    .execute();
```

File: scenarios/k8s-scaling/rules.drl [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+             cronChecker.checkPatternForSession("* */3 * ? * * *"))
+    then
+        kubernetesService = usableServices.getKubernetesService();
+        HorizontalScalingAction.builder()
+                .kubernetesService(kubernetesService)
+                .namespace("sock-shop")
+                .resourceType("Deployment")
+                .resourceName("front-end")
+                .replicas(-1)
+                .build();
```
Comment (proman3419) [inline scenarios/k8s-scaling/rules.drl:44]:
```suggestion
                .build()
                .execute();
```

File: scenarios/k8s-scaling/rules.drl [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+             cronChecker.checkPatternForSession("* */2 * ? * * *"))
+    then
+        kubernetesService = usableServices.getKubernetesService();
+        HorizontalScalingAction.builder()
+                    .kubernetesService(kubernetesService)
+                    .namespace("sock-shop")
+                    .resourceType("Deployment")
+                    .resourceName("user")
+                    .replicas(1)
+                    .build();
```
Comment (proman3419) [inline scenarios/k8s-scaling/rules.drl:63]:
```suggestion
                    .build()
                    .execute();
```

File: scenarios/k8s-scaling/rules.drl [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+             cronChecker.checkPatternForSession("* */2 * ? * * *"))
+    then
+        kubernetesService = usableServices.getKubernetesService();
+        HorizontalScalingAction.builder()
+                    .kubernetesService(kubernetesService)
+                    .namespace("sock-shop")
+                    .resourceType("Deployment")
+                    .resourceName("user")
+                    .replicas(-1)
+                    .build()
```
Comment (proman3419) [inline scenarios/k8s-scaling/rules.drl:82]:
```suggestion
                    .build()
                    .execute();
```

File: scenarios/k8s-scaling/run.sh [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# shellcheck disable=SC2046
+echo $(date +"%Y-%m-%d %T") start test
+
+curr=0
+for ((i=1; i<=6; i++)); do
+    ((curr++))
+    echo "$(date +"%Y-%m-%d %T") Scaling loadtest to $curr"
+    kubectl scale deploy load-test -n loadtest --replicas="$curr"
+
```
Comment (proman3419) [inline scenarios/k8s-scaling/run.sh:9]: Remove newlines

File: scenarios/k8s-scaling/run.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# shellcheck disable=SC2046
+echo $(date +"%Y-%m-%d %T") start test
```
Comment (proman3419) [inline scenarios/k8s-scaling/run.sh:2]:
```suggestion
echo "$(date +'%Y-%m-%d %T') start test"
```
Enclosing in " is not required? :o

File: scenarios/k8s-scaling/run.sh [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    kubectl scale deploy load-test -n loadtest --replicas="$curr"
+
+
+    sleep 180
+done
+sleep 240
+echo scaling down
+kubectl scale deploy load-test -n loadtest --replicas=0
+sleep 60
+echo $(date +"%Y-%m-%d %T") test finished
```
Comment (proman3419) [inline scenarios/k8s-scaling/run.sh:17]:
```suggestion
echo "$(date +'%Y-%m-%d %T') test finished"
```

**Review by proman3419 — CHANGES_REQUESTED** [review]

File: scenarios/k8s-scaling/cm.yml (hunk 2 above)
Comment (Agwen12) [reply scenarios/k8s-scaling/cm.yml:1]: it is but when you actualy want to try it out it's mych easier to just run `kubectl apply -f cm.yaml ` then coping it yourself

File: scenarios/k8s-scaling/run.sh (hunk 8 above)
Comment (Agwen12) [reply scenarios/k8s-scaling/run.sh:2]: nope

File: scenarios/k8s-scaling/cm.yml (hunk 2 above)
Comment (proman3419) [reply scenarios/k8s-scaling/cm.yml:1]:
```
make upload-rules SCENARIO_NAME=k8s-scaling
```
does exactly that, no need to keep this file

Comment (Agwen12) [reply scenarios/k8s-scaling/cm.yml:1]: ehhh, okeeeey then

**Review by proman3419 — APPROVED** [review]

**Pull request merged**